packages/wrangler/src/__tests__/configuration.pages.test.ts

```diff
@@ -0,0 +1,846 @@
+			"Processing wrangler configuration:
+			  - Hyperdrive Bindings are currently in beta to allow the API to evolve before general availability.
+			    Please report any issues to https://github.com/cloudflare/workers-sdk/issues/new/choose
+			  - In wrangler.toml, you have configured [durable_objects] exported by this Worker (TEST_DO_CLASS), but no [migrations] for them. This may not work as expected until you add a [migrations] section to your wrangler.toml. Add this configuration to your wrangler.toml:
+
+			      \`\`\`
+			      [[migrations]]
+			      tag = \\"v1\\" # Should be unique for each entry
+			      new_classes = [\\"TEST_DO_CLASS\\"]
+			      \`\`\`
+
+			    Refer to https://developers.cloudflare.com/workers/learning/using-durable-objects/#durable-object-migrations-in-wranglertoml for more details."
+		`);
+				expect(diagnostics.hasErrors()).toBe(false);
+				expect(config).toEqual(
+					expect.objectContaining({
+						/** TOP-LEVEL ONLY FIELDS **/
+						pages_build_output_dir: "./public",
+						dev: {
+							ip: "localhost",
+							port: 1234,
+							inspector_port: 5678,
+							local_protocol: "https",
+							upstream_protocol: "https",
+							host: "127.0.0.0",
+						},
+						send_metrics: true,
+
+						/** INHERITABLE ENVIRONMENT FIELDS **/
+						name: "pages-is-awesome",
+						compatibility_date: "2024-01-01",
+						compatibility_flags: ["COMPATIBILITY_FLAG"],
+						limits: { cpu_ms: 100 },
+						placement: { mode: "smart" },
+
+						/** NON-INHERITABLE ENVIRONMENT FIELDS **/
+						vars: { VAR: "TEST_VAR" },
+						durable_objects: {
+							bindings: [
+								{ name: "TEST_DO_BINDING", class_name: "TEST_DO_CLASS" },
+							],
+						},
+						kv_namespaces: [
+							{
+								binding: "TEST_KV_BINDING",
+								id: "TEST_KV_ID",
+							},
+						],
+						queues: {
+							producers: [
+								{
+									binding: "TEST_QUEUE_PRODUCER_BINDING",
+									queue: "TEST_QUEUE_PRODUCER_NAME",
+								},
+							],
+						},
+						r2_buckets: [
+							{
+								binding: "TEST_R2_BINDING",
+								bucket_name: "TEST_R2_BUCKET_NAME",
+							},
+						],
+						d1_databases: [
+							{
+								binding: "TEST_D1_BINDING",
+								database_id: "TEST_D1_DB_ID",
+								database_name: "TEST_D1_DB_NAME",
+							},
+						],
+						vectorize: [
+							{
+								binding: "TEST_VECTORIZE_BINDING",
+								index_name: "TEST_VECTORIZE_INDEX_NAME",
+							},
+						],
+						hyperdrive: [
+							{
+								binding: "TEST_HYPERDRIVE_BINDING",
+								id: "TEST_HYPERDRIVE_ID",
+							},
+						],
+						services: [
+							{
+								binding: "TEST_SERVICE_BINDING",
+								service: "TEST_SERVICE_NAME",
+							},
+						],
+						analytics_engine_datasets: [
+							{
+								binding: "TEST_AED_BINDING",
+							},
+						],
+						ai: {
+							binding: "TEST_AI_BINDING",
+						},
+					})
+				);
+
+				/*======================================*/
```

Comment:
just calling this test out to highlight that this happens because we do the Pages-specific validation **after* `normalizeAndValidateConfig()`

Reply:
// TODO @CarmenPopoviciu add a comment to explain the above, if we decide to keep this test as part of the suite

Reply:
comments added ✅